src/atomicStore.ts

```diff
@@ -0,0 +1,206 @@
+    : T[K];
+};
+
+/** Generated store type where each property becomes a Jotai atom */
+type AtomicStore<T> = {
+  [K in keyof T]: T[K] extends (...args: any[]) => any
+    ? WritableAtom<void, Parameters<T[K]>, void>
+    : K extends WritableKeys<T>
+      ? PrimitiveAtom<T[K]>
+      : Atom<T[K]>;
+};
+
+const ACTION = Symbol('ACTION'); // Unique symbol to denote action atoms
+
+/**
+ * Creates an atomic store that combines Zustand-like state definition with Jotai atoms.
+ *
+ * The state definition object passed to this function can consist of three kinds of properties:
+ *
+ * **Base State**
+ *    - Regular properties (not functions or getters)
+ *    - Stored in a single root atom for efficient updates
+ *    - Becomes a `PrimitiveAtom<Value>` in the store
+ *
+ * **Derived State**
+ *    - Defined using property getters
+ *    - Auto-updates when dependencies change
+ *    - Cached by Jotai to prevent unnecessary recomputation
+ *    - Becomes a read-only `Atom<Value>` in the store
+ *
+ * **Actions**
+ *    - Regular functions that can update state
+ *    - Can return partial `Partial<State>` updates or modify state via `this`
+ *    - Becomes a `WritableAtom<void, Args, void>` in the store
+ *
+ * @returns An object where each property is converted to a Jotai atom:
+ *    - Base state becomes a primitive atom: `PrimitiveAtom<Value>`
+ *    - Derived state becomes a read-only atom: `Atom<Value>`
+ *    - Actions become writable atoms: `WritableAtom<void, Args, void>`
+ *
+ * @example Basic store definition
+ * ```ts
+ * const store = createAtomicStore({
+ *   count: 0,
+ *   get double() { return this.count * 2 },
+ *   increment(n = 1) { return { count: this.count + n } }
+ * })
+ * ```
+ *
+ * @example Usage with React hooks
+ * ```tsx
+ * function Counter() {
+ *   // Read base or derived state with useAtomValue
+ *   const count = useAtomValue(store.count)
+ *   const double = useAtomValue(store.double)
+ *
+ *   // Get action setter with useSetAtom
+ *   const increment = useSetAtom(store.increment)
+ *
+ *   return (
+ *     <div>
+ *       <p>Count: {count}</p>
+ *       <p>Double: {double}</p>
+ *       <button onClick={() => increment()}>+1</button>
+ *       <button onClick={() => increment(5)}>+5</button>
+ *     </div>
+ *   )
+ * }
+ * ```
+ *
+ * @example Direct usage with Jotai store
+ * ```ts
+ * const jotai = createStore()
+ *
+ * // Read values
+ * const count = jotai.get(store.count)
+ * const double = jotai.get(store.double)
+ *
+ * // Update base state
+ * jotai.set(store.count, 42)
+ *
+ * // Call actions
+ * jotai.set(store.increment)
+ * ```
+ *
+ * @template State - Type of the state definition object
+ */
+export function createAtomicStore<State extends object>(
+  definition: AtomicDefinition<State>,
+): AtomicStore<State> {
+  const store = {} as AtomicStore<State>;
+  const baseAtoms = new Map<keyof State, PrimitiveAtom<any>>();
+
+  // Helper to check property types
+  const isPropType = {
```

Comment:
Just a nit:
Minification doesn't work for properties, so I'd define three functions:
```js
const isActionPropType = (key) => ...;
const isDerivedPropType = (key) => ...;
const isBasePropType = (key) => ...;
```